modules/vaos/app/services/vaos/user_service.rb

```diff
@@ -2,37 +2,103 @@
 
 module VAOS
   class UserService < VAOS::BaseService
-    def session
-      cached = SessionStore.find(user.account_uuid)
+    def session(user)
+      cached = cached_by_account_uuid(user.account_uuid)
       return cached.token if cached
 
-      token = get_session_token
-      session_store = SessionStore.new(account_uuid: user.account_uuid, token: token)
-      session_store.save
-      session_store.expire(get_ttl(token))
+      new_session_token(user)
+    end
 
-      token
+    def extend_session(account_uuid)
+      unless session_creation_locked?(account_uuid)
+        lock_session_creation(account_uuid)
+        ExtendSessionJob.perform_async(account_uuid)
+      end
+    end
+
+    def update_session_token(account_uuid)
+      cached = cached_by_account_uuid(account_uuid)
+      if cached
+        url = '/users/v2/session/jwts'
+        response = perform(:get, url, nil, refresh_headers(account_uuid))
+        new_token = response.body[:jws]
+        Rails.logger.info('VAOS session updated',
+                          {
+                            account_uuid: account_uuid,
```

Comment:
NITPICK:

I was going to suggest moving this into save_session!  but I think you're using the same method for refresh and first time creation, so i guess this works. Trying to think if maybe save_session could save_and_log_session but accepts parameters to inform which type of logging to do. 

in other words:

```ruby
def save_session!(action = 'update')
   ... do stuff
   if success 
      "VAOS session #{action} successful"
   else
      "VAOS session #{action} failed"
  end
end